rkt/config/directories.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+type configurableDirectoriesV1 struct {
+	Data string `json:"data"`
+}
+
+func init() {
+	addParser("directories", "v1", &configurableDirectoriesV1{})
+	// Look in 'directories' subdir for configs of type directories
+	registerSubDir("dirs.d", []string{"directories"})
+}
+
+func (p *configurableDirectoriesV1) parse(config *Config, raw []byte) error {
+	var dirs configurableDirectoriesV1
+	if err := json.Unmarshal(raw, &dirs); err != nil {
+		return err
+	}
+	if config.Directories.DataDir != "" {
```

Comment:
To be future-proof, I would put this under the `if dirs.Data != ""` clause. It works currently only because there is one setting in this kind. But if we add another field (say `run`) and then put two config files of `directories` kind in the same directory, `data.json` defining only `data` field and `run.json` defining only `run` field, this code will bail out if `data.json` is parsed before `run.json`.